### PR TITLE
Show date picker option for Image section

### PR DESCRIPTION
**Pull request by srgzlb**
The designs for the Tough and Romanic versions both show the Date picker as part of the Images section. This option was removed after the addition of a separate Date picker section but should be restored.

Changelog: 
1. Re-added `Show date picker` option to `Image` section (like the `Hero` section)
image
image

2. Fixed typo of `Show datepicker` label to `Show date picker` in schema in the `Hero` section
Google Chrome_2023-10-09 15-35-27@2x

3. Slightly refactored the `Date picker` in `Hero` section
4. Added auto counting height of `Date picker` function and pushed it to css variable
5. Added function of auto change the color of `Datepicker` title regarding the background of the section
Google Chrome_2023-10-09 15-37-19@2x
Google Chrome_2023-10-09 15-38-29@2x

6. Removed an old `Date picker` section
7. Templates were corrected as well

### CODE REVIEW

File: assets/carousel.js [hunk 1]
```diff
@@ -456,29 +462,36 @@ class Carousel {
...
-    !isOld && !isNew
-      ? element.classList.add(addClass)
-      : element.classList.replace(oldClass, newClass)
+    let isOld, isNew;
+
+    Object.values(elements).forEach(element => {
+      if (!element) return false;
+
+      isOld = element.classList.contains(oldClass),
+      isNew = element.classList.contains(newClass);
```
Comment (zmajstor) [inline assets/carousel.js:489]:
```suggestion
      isOld = element.classList.contains(oldClass)
      isNew = element.classList.contains(newClass)
```

File: assets/carousel.js [hunk 2]
```diff
@@ -456,29 +462,36 @@ class Carousel {
...
       }
     })
   }
 
-  setClass(element, options) {
-    const {addClass, oldClass, newClass} = options,
-          isOld = element.classList.contains(oldClass),
-          isNew = element.classList.contains(newClass);
+  setClass(elements, options) {
+    const {addClass, oldClass, newClass} = options;
```
Comment (zmajstor) [inline assets/carousel.js:481]:
```suggestion
    const { addClass, oldClass, newClass } = options
```

File: assets/carousel.js [hunk 3]
```diff
@@ -456,29 +462,36 @@ class Carousel {
...
         oldClass: this.classes.showLight,
         newClass: this.classes.showDark
-      }
+      };
 
       if (itemIndex + 1 === index) {
-        if (isLight) this.setClass(this.wrap, optionsLight)
-        if (isDark) this.setClass(this.wrap, optionsDark)
+        if (isLight) this.setClass(elements, optionsLight);
+        if (isDark) this.setClass(elements, optionsDark);
```
Comment (zmajstor) [inline assets/carousel.js:475]:
```suggestion
        if (isLight) this.setClass(elements, optionsLight)
        if (isDark) this.setClass(elements, optionsDark)
```

File: assets/carousel.js [hunk 4]
```diff
@@ -505,14 +518,25 @@ class Carousel {
...
-        sibling = sibling.previousElementSibling;
-      }
+    let sibling;
+
+    const setSibling = () => {
+      sibling = direction === this.event.prev
+                ? element.previousElementSibling
+                : direction === this.event.next
+                ? element.nextElementSibling
+                : false
```
Comment (zmajstor) [inline assets/carousel.js:531]: this is hard to understand. I think this is better:

```suggestion
      let sibling = false;

      if (direction === this.event.prev) {
        sibling = element.previousElementSibling;
      } else if (direction === this.event.next) {
        sibling = element.nextElementSibling;
      }
```

**Review by zmajstor — CHANGES_REQUESTED** [review]
nice, just a few javascript suggestions

File: assets/carousel.js (hunk 4 above)
Comment (srgzlb) [reply assets/carousel.js]: Yeah, it was slightly convoluted

**Pull request merged**